chef/cookbooks/monasca/recipes/metric_agent.rb

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright 2016 SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package "openstack-monasca-agent"
+
+agent_settings = node[:monasca][:metric_agent]
+agent_keystone = agent_settings[:keystone]
+
+keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
+
+monasca_servers = search(:node, "roles:monasca-server") || []
+monasca_server = monasca_servers[0]
+
+ha_enabled = node[:monasca][:ha][:enabled]
+ssl_enabled = node[:monasca][:api][:ssl]
+monasca_protocol = ssl_enabled ? "https" : "http"
+monasca_public_host = CrowbarHelper.get_host_for_public_url(
+  monasca_server, ssl_enabled, ha_enabled
+)
+
+agent_dimensions = { service: "monitoring" }
+service = "monitoring"
+monasca_url = "#{monasca_protocol}://#{monasca_public_host}/v3"
+
+template "/etc/sudoers.d/monasca-agent" do
```

Comment:
Well, I did not found any `sudo` in rpm-packaging however I found it in crowbar-openstack...:
https://github.com/crowbar/crowbar-openstack/blob/master/chef/cookbooks/nova/recipes/monitor.rb#L50-L55